app/views/ChooseProvider.js

```diff
@@ -249,88 +262,56 @@ class ChooseProviderScreen extends Component {
             )}
           </View>
 
-          <View style={styles.listContainer}>
-            {Object.keys(this.state.selectedAuthorities).length == 0 ? (
-              <>
+          <SPKeyboardAvoidingView>
+            <View style={styles.listContainer}>
+              {this.state.selectedAuthorities.length == 0 &&
+              !this.state.urlEntryInProgress ? (
                 <Typography
                   style={[styles.sectionDescription, styles.noDataSourceText]}
                   use={'headline2'}>
                   {languages.t('label.authorities_no_sources')}
                 </Typography>
-                <View
-                  style={[
-                    styles.flatlistRowView,
-                    { display: this.state.displayUrlEntry },
-                  ]}>
-                  <DynamicTextInput
-                    onChangeText={text => {
-                      this.setState({
-                        urlText: text,
-                      });
-                    }}
-                    value={this.state.urlText}
-                    autoFocus={this.state.urlEntryInProgress}
-                    style={[styles.item, styles.textInput]}
-                    placeholder={languages.t(
-                      'label.authorities_input_placeholder',
+              ) : (
+                <>
+                  <View
+                    style={[
+                      styles.flatlistRowView,
+                      { display: this.state.displayUrlEntry },
+                    ]}>
+                    <DynamicTextInput
+                      onChangeText={text => {
+                        this.setState({
+                          urlText: text,
+                        });
+                      }}
+                      value={this.state.urlText}
+                      autoFocus={this.state.urlEntryInProgress}
+                      style={[styles.item, styles.textInput]}
+                      placeholder={languages.t('label.enter_authority_url')}
+                      onSubmitEditing={this.addCustomUrlToState}
+                    />
+                    <TouchableOpacity onPress={this.addCustomUrlToState}>
```

Comment:
This is almost an IconButton but with an image. Can you modify IconButton to include image={img} and use that?

Reply:
IconButton seems to be a styled SvgXml, and it won't accept png images. So we might need to find svg alternative for these pngs or bring in another lib like react-native-vector-icons that does the transformation for you

Reply:
future work.